crates/next-error-code-swc-plugin/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+use std::{
+    collections::{hash_map::DefaultHasher, HashMap},
+    fs,
+    hash::{Hash, Hasher},
+};
+
+use swc_core::{
+    ecma::{ast::*, transforms::testing::test_inline, visit::*},
+    plugin::{plugin_transform, proxies::TransformPluginProgramMetadata},
+};
+
+pub struct TransformVisitor {
+    commit_hash: String,
+    file_path: String,
+    mode: String,
```

Comment:
wouldn't it be better to make this an enum? that'd save us the `panic!` in the loop
```
enum Mode { Generate, Check }
```